ui/app/AppLayouts/Wallet/components/SendModalContent.qml

```diff
@@ -91,33 +91,37 @@ Item {
         }
     }
 
-    Input {
-        id: txtAmount
-        //% "Amount"
-        label: qsTrId("amount")
-        anchors.top: parent.top
-        //% "Enter amount..."
-        placeholderText: qsTrId("enter-amount...")
-        validationError: amountValidationError
-    }
-
-    AssetSelector {
-        id: selectAsset
-        assets: walletModel.assets
-        anchors.top: txtAmount.bottom
-        anchors.topMargin: Style.current.padding
-        anchors.right: parent.right
-        width: 86
-        height: 28
+    AssetAndAmountInput {
+      id: txtAmount
+      assets: walletModel.assets
+      defaultCurrency: walletModel.defaultCurrency
+      errorMessage: amountValidationError
+      anchors.top: parent.top
+      onInputChanged: {
+        if (isNaN(input)) {
+          return
+        }
+        if (input === "") {
+          txtAmount.fiatBalance = "0.00"
+        } else {
+          txtAmount.fiatBalance = walletModel.getFiatValue(input, assetSymbol, currency)
+        }
+      }
+      onFiatValueChanged: {
+        txtAmount.text = walletModel.getCryptoValue(input, currency, assetSymbol)
+      }
     }
 
     AccountSelector {
         id: selectFromAccount
         accounts: walletModel.accounts
-        anchors.top: selectAsset.bottom
+        anchors.top: txtAmount.bottom
         anchors.topMargin: Style.current.padding
         anchors.left: parent.left
         anchors.right: parent.right
+        onAccountSelected: function (selectedAccount, index) {
+          walletModel.setCurrentAccountByIndex(index)
```

Comment:
Instead of keeping track of the selected wallet account in the model, maybe it would be better to set a property on the component:
```
txtAmount.selectedAccount = selectedAccount
```

In that case, we can get rid of the new signal and use the built-in `onSelectedAccountChanged`.

Reply:
I think this is another case where we have to decide who's responsible to provide a component with data. The reason I have to do `walletModel.setCurrentAccountByIndex()` is because we need the `AssetSelector` to update when the account is changing. Unfortunately, we can't rely on `selectedAccount` here because there are no assets attached to it. The `AssetSelector` receives its assets via `assets: walletModel.assets`.

^ That property is updated every time the `walletModel.currentAccount` has changed. So unless we attach `assets` to the currently `selectedAccount` (which may or may not be possible with Qt -> QML), we have to rely on `walletModel.currentAccount` to change which will trigger a change in `walletModel.assets`.

Hope this makes sense

Reply:
Each account should have assets associated with. See https://github.com/status-im/nim-status-client/blob/e484c412910800e0b3060ed8254d2c1a46244432/ui/shared/AccountSelector.qml#L23 for an example.

Reply:
Setting the account in the view is currently used when switching between accounts in the wallet, so setting this property might interfere, or might inadvertently switch between wallet accounts.

Reply:
For transparency: @emizzle and I had a call about this and implemented it. Assets are not derived from the currently selected account